Content.Server/Mobs/Mind.cs

```diff
@@ -272,8 +275,14 @@ public void TransferTo(IEntity? entity)
             OwnedComponent = component;
             OwnedComponent?.InternalAssignMind(this);
 
-            if (VisitingEntity?.HasComponent<GhostComponent>() == false)
+            GhostComponent? ghostComponent = null;
+
+            if (ghostCheckOverride || // to force mind transfer, for example from ControlMobVerb
+                (VisitingEntity?.TryGetComponent(out ghostComponent) ?? false) || // visiting entity is not a Ghost
+                (ghostComponent != null && !ghostComponent.CanReturnToBody))  // it is a ghost, but cannot return to body anyway, so it's okay                 
```

Comment:
This looks a bit off. Try this:
```suggestion
            if (IsVisitingEntity
                && (ghostCheckOverride // to force mind transfer, for example from ControlMobVerb
                || !VisitingEntity.TryGetComponent(out GhostComponent? ghostComponent) // visiting entity is not a Ghost
                || !ghostComponent.CanReturnToBody))  // it is a ghost, but cannot return to body anyway, so it's okay
```
Might need  a not null marker. `!VisitingEntity!.TryGetComponent(out ghostComponent)`